src/plugins/embeddable/public/embeddable_panel/panel_header/use_embeddable_panel_badges.tsx

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Subscription } from 'rxjs';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { EuiBadge, EuiNotificationBadge, EuiToolTip } from '@elastic/eui';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import {
+  IEmbeddable,
+  panelBadgeTrigger,
+  panelNotificationTrigger,
+  PANEL_BADGE_TRIGGER,
+  PANEL_NOTIFICATION_TRIGGER,
+} from '../..';
+import { uiActions } from '../../kibana_services';
+import {
+  EmbeddableBadgeAction,
+  EmbeddableNotificationAction,
+  EmbeddablePanelProps,
+} from '../types';
+
+export const useEmbeddablePanelBadges = (
+  embeddable: IEmbeddable,
+  getActions: EmbeddablePanelProps['getActions']
+) => {
+  const getActionsForTrigger = getActions ?? uiActions.getTriggerCompatibleActions;
+
+  const [badges, setBadges] = useState<EmbeddableBadgeAction[]>();
+  const [notifications, setNotifications] = useState<EmbeddableNotificationAction[]>();
+
+  const getAllBadgesFromEmbeddable = useCallback(async () => {
+    let currentBadges: EmbeddableBadgeAction[] =
+      ((await getActionsForTrigger(PANEL_BADGE_TRIGGER, {
+        embeddable,
+      })) as EmbeddableBadgeAction[]) ?? [];
+
+    const { disabledActions } = embeddable.getInput();
+    if (disabledActions) {
+      currentBadges = currentBadges.filter((badge) => disabledActions.indexOf(badge.id) === -1);
+    }
+    return currentBadges;
+  }, [embeddable, getActionsForTrigger]);
+
+  const getAllNotificationsFromEmbeddable = useCallback(async () => {
+    let currentNotifications: EmbeddableNotificationAction[] =
+      ((await getActionsForTrigger(PANEL_NOTIFICATION_TRIGGER, {
+        embeddable,
+      })) as EmbeddableNotificationAction[]) ?? [];
+
+    const { disabledActions } = embeddable.getInput();
+    if (disabledActions) {
+      currentNotifications = currentNotifications.filter(
+        (badge) => disabledActions.indexOf(badge.id) === -1
+      );
+    }
+    return currentNotifications;
+  }, [embeddable, getActionsForTrigger]);
+
+  /**
+   * On embeddable creation get initial badges & notifications then subscribe to all
+   * input updates to refresh them
+   */
+  useEffect(() => {
+    let mounted = true;
+    let subscription: Subscription;
+
+    const updateNotificationsAndBadges = async () => {
+      const [newBadges, newNotifications] = await Promise.all([
+        getAllBadgesFromEmbeddable(),
+        getAllNotificationsFromEmbeddable(),
+      ]);
+      if (!mounted) return;
+      setBadges(newBadges);
+      setNotifications(newNotifications);
+    };
+
+    updateNotificationsAndBadges().then(() => {
+      if (mounted) {
```

Comment:
Also check for `canceled` to avoid creating subcriptions if useEffect is fired before previous async action completes

Reply:
This variable actually is tracking the `canceled` state - you can see that on line 99 - I've just called it `mounted` here instead. I'll rename the variable to be more consistent though!